watch/watch.go

```diff
@@ -0,0 +1,360 @@
+// Copyright 2017 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package watch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage"
+)
+
+// Watcher allows for watches to be registed on queries. When a dependency of a watched
+// query changes, a notification will be send back to the user.
+type Watcher struct {
```

Comment:
I'm not sure what else it would look like, the "watch" is really just a goroutine. The watcher starts up the goroutine, and then it's only responsibility is to shut it down once the watch is ended.